servicetalk-utils-internal/src/main/java/io/servicetalk/utils/internal/ThrowableUtils.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.utils.internal;
+
+import javax.annotation.Nullable;
+
+/**
+ * Utilities for working with {@link Throwable}s.
+ */
+public final class ThrowableUtils {
```

Comment:
In a follow up PR I'm going to consolidate in this class all utilities related to `Throwable`:
1. `PlatformDependent.throwException(Throwable)`;
2. `io.servicetalk.concurrent.internal.ThrowableUtils`.

It's deferred to allow backward compatible cherry-picking for 0.41.